lib/ruby_smb/client/negotiation.rb

```diff
@@ -140,6 +140,7 @@ def parse_negotiate_response(packet)
           self.server_guid = packet.server_guid
           self.server_start_time = packet.server_start_time.to_time if packet.server_start_time != 0
           self.server_system_time = packet.system_time.to_time if packet.system_time != 0
+          self.server_supports_multi_credit = packet&.capabilities&.large_mtu == 1
```

Comment:
If you checked the dialect here, you wouldn't need to evaluate it later on which would simplify the use.

```suggestion
          self.server_supports_multi_credit = self.dialect != '0x0202' && packet&.capabilities&.large_mtu == 1
```

I think it would make more sense if dialect 0x0202 does not support multi credit.

An example that could then be simplified:
https://github.com/rapid7/ruby_smb/pull/162/files#diff-1faf31d70004f38df4567742a7efd48fR119

Reply:
Yeah, it makes more sense. I will update this.